### DISCUSSION TITLE
AAPS no longer accepts BG values from Xdrip+ for SMB

### DISCUSSION THREAD

**Discussion by kropas13** [NightscoutFoundation/xDrip#3278] [Bug reports] [answered]
After having updated xDrip to the latest nightly (version 3b418f2-2024.01.05), AAPS no longer accepts the transmitted BG values for SMBs (Safety warning in AAPS/SMB: SMB always and after carbs disabled because active BG source doesn't support advanced filtering). 

Used software/devices
- AAPS v3.2.0.3
- Pixel 5
- CGM Dexcom G7

No other parameter has been modified neither in xDrip nor in AAPS after the update.
As I understand, AAPS requires raw (unfiltered) BG data which it smooths with its own internal filters to be able to deliver SMBs.

**Comment by Navid200** [comment]
Would you please confirm that when you use this version of xDrip instead, the problem goes away?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.01.02

**Reply by kropas13** [reply]
Yes, indeed, this version works!

**Reply by Navid200** [reply, marked as answer]
Thanks

Would you please communicate this with AAPS developers if you haven't already?

The changes in the last xDrip release are cosmetic.  I have changed a few labels, in xDrip, to say G7 instead of G6.  When G7 was first added to xDrip, everything looked like we were still using a G6.  This was done to expedite the release so that users could use G7 as soon as possible.
I have more of these changes coming.  But, they are phased out gradually so that we can see problems like this.  Thanks for reporting this right away.

If an AAPS developer can tell me what change has caused this problem, I may be able to fix it.  Alternatively, it is possible that something in AAPS may need to change as well to accommodate the G7 label.
Otherwise, this will have to wait for our lead developer to have a look at, which may not happen right away.

**Reply by kropas13** [reply]
Thanks for your quick answer and solution. 
I guess AAPS does not yet identify the G7 label as a reliable BG source, hence the issue.
I'll post a bug report on AAPS.

**Comment by olorinmaia** [comment]
Milos has now closed issue within AAPS. Try build latest dev32 and see if it solved this issue

https://github.com/nightscout/AndroidAPS/issues/3160?fbclid=IwAR2z3tfmJu_pkgsdpeMzwK0TWOgyvpKjl_R0eYlfbhHg1fDHSaRCOpThIhM

**Comment by Navid200** [comment]
@kropas13 What is the status of this?
If you are still experiencing this problem, would you please update xDrip to the latest Nightly and tell us if it still fails?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.01.13

**Reply by kropas13** [reply]
yes, the problem remains with the latest nightly. But it seems that Milos has fixed it in AAPS (I haven't tested yet).

**Reply by jamorham** [reply]
Can you test the latest xDrip nightly, it makes G7 look like G6 so should be accepted by older AAPS.

**Reply by kropas13** [reply]
This works, thanks
